app/services/train_api.rb

```diff
@@ -10,11 +10,20 @@ def index
     JSON.parse(response.body)
   end
 
+  def new
+    response = self.class.get('/v1/trains/new')
+    JSON.parse(response.body)
+  end
```

Comment:
this wont work, as you have said the api doesnt have a new end point and it doesnt need one.
You can remove all this entirely.